### ISSUE TITLE
Fails to massage pnpm release notes

### ISSUE THREAD

**Issue by ylemkimon** [renovatebot/renovate#16921]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Renovate fails to massage issue links in the `What's Changed` section of pnpm release notes, leading to autolinks, e.g., https://github.com/renovatebot/docker-buildpack/pull/338.

It seems to be caused by an HTML table; the raw release note is as follows:

### Relevant debug logs

<details><summary>Raw Release Note</summary>

```markdown
## Minor Changes

-   A new setting supported: `prefer-symlinked-executables`. When `true`, pnpm will create symlinks to executables in
    `node_modules/.bin` instead of command shims (but on POSIX systems only).

    This setting is `true` by default when `node-linker` is set to `hoisted`.

    Related issue: [#4782](https://github.com/pnpm/pnpm/issues/4782).

-   When `lockfile-include-tarball-url` is set to `true`, every entry in `pnpm-lock.yaml` will contain the full URL to the package's tarball [#5054](https://github.com/pnpm/pnpm/pull/5054).

## Patch Changes

-   `pnpm deploy` should include all dependencies by default [#5035](https://github.com/pnpm/pnpm/issues/5035).
-   Don't print warnings about file verifications. Just print info messages instead.
-   `pnpm publish --help` should print the `--recursive` and `--filter` options [#5019](https://github.com/pnpm/pnpm/issues/5019).
-   It should be possible to run exec/run/dlx with the `--use-node-version` option.
-   `pnpm deploy` should not modify the lockfile [#5071](https://github.com/pnpm/pnpm/issues/5071)
-   `pnpm deploy` should not fail in CI [#5071](https://github.com/pnpm/pnpm/issues/5071)
-   When `auto-install-peers` is set to `true`, automatically install direct peer dependencies [#5028](https://github.com/pnpm/pnpm/pull/5067).

    So if your project the next manifest:

    ```json
    {
      "dependencies": {
        "lodash": "^4.17.21"
      },
      "peerDependencies": {
        "react": "^18.2.0"
      }
    }
    ```

    pnpm will install both lodash and react as a regular dependencies.

## Our Gold Sponsors

<table>
  <tbody>
    <tr>
      <td align="center" valign="middle">
        <a href="https://bit.dev/?utm_source=pnpm&utm_medium=release_notes" target="_blank"><img src="https://raw.githubusercontent.com/pnpm/pnpm.github.io/main/static/img/users/bit.svg" width="80"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://nhost.io/?utm_source=pnpm&utm_medium=release_notes" target="_blank"><img src="https://raw.githubusercontent.com/pnpm/pnpm.github.io/main/static/img/users/nhost.svg" width="180"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://novu.co/?utm_source=pnpm&utm_medium=release_notes" target="_blank"><img src="https://raw.githubusercontent.com/pnpm/pnpm.github.io/main/static/img/users/novu.svg" width="180"></a>
      </td>
    </tr>
  </tbody>
</table>

## Our Silver Sponsors

<table>
  <tbody>
    <tr>
      <td align="center" valign="middle">
        <a href="https://prisma.io/?utm_source=pnpm&utm_medium=release_notes" target="_blank"><img src="https://raw.githubusercontent.com/pnpm/pnpm.github.io/main/static/img/users/prisma.svg" width="180"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://leniolabs.com/?utm_source=pnpm&utm_medium=release_notes" target="_blank"><img src="https://raw.githubusercontent.com/pnpm/pnpm.github.io/main/static/img/users/leniolabs.jpg" width="80"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://vercel.com/?utm_source=pnpm&utm_medium=release_notes" target="_blank"><img src="https://raw.githubusercontent.com/pnpm/pnpm.github.io/main/static/img/users/vercel.svg" width="180"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://www.takeshape.io/?utm_source=pnpm&utm_medium=release_notes" target="_blank"><img src="https://raw.githubusercontent.com/pnpm/pnpm.github.io/main/static/img/users/takeshape.svg" width="280"></a>
      </td>
    </tr>
  </tbody>
</table>

## What's Changed
* pnpm rebuild accepts --store-dir by @chengcyber in https://github.com/pnpm/pnpm/pull/5036
* fix(deploy): include all deps by default by @zkochan in https://github.com/pnpm/pnpm/pull/5040
* chore(deps): upgrade nock to v13 by @mcmxcdev in https://github.com/pnpm/pnpm/pull/5043
* fix: log more info on HTTP error by @zkochan in https://github.com/pnpm/pnpm/pull/4917
* fix: document the -r option by @zkochan in https://github.com/pnpm/pnpm/pull/5044
* chore(deps): upgrade sinon to v14 by @mcmxcdev in https://github.com/pnpm/pnpm/pull/5045
* fix(audit): add authentication to pnpm-audit by @sled in https://github.com/pnpm/pnpm/pull/5053
* feat: prefer-symlinked-executables by @zkochan in https://github.com/pnpm/pnpm/pull/5048
* chore: update pnpm-workspace.yaml by @ayu14214 in https://github.com/pnpm/pnpm/pull/5060
* feat: add `lockfile-include-tarball-url` option by @MBelniak in https://github.com/pnpm/pnpm/pull/5054
* fix: auto install root peer deps when auto-install-peers=true by @zkochan in https://github.com/pnpm/pnpm/pull/5067
* fix(deploy): don't modify the lockfile and fail in CI by @zkochan in https://github.com/pnpm/pnpm/pull/5074

## New Contributors
* @mcmxcdev made their first contribution in https://github.com/pnpm/pnpm/pull/5043
* @sled made their first contribution in https://github.com/pnpm/pnpm/pull/5053
* @ayu14214 made their first contribution in https://github.com/pnpm/pnpm/pull/5060
* @MBelniak made their first contribution in https://github.com/pnpm/pnpm/pull/5054

**Full Changelog**: https://github.com/pnpm/pnpm/compare/v7.5.2...v7.6.0
```

</details>


### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by HonkingGoose** [comment]
Here's our reproduction! 😉 

[`renovate-reproductions/16921` on GitHub](https://github.com/renovate-reproductions/16921)

**Comment by MaronHatoum** [comment]
After investigating and debugging this issue, seems like there are some '\n' missing when getting release notes by 'getChangelogs()'.
I added a new regex in `sanitizeMarkdown()` to detect the missing '\n' and add them 
```
  res = res.replace(regEx(/(?<before>[^\n]\n)(?<title>####.*)/g), '$<before>\n$<title>');
```
as a result of this change, I found another issue in:https://github.com/renovatebot/renovate/blob/5a9f2cdb0297d73348dcc9977ce2c0403b00245e/lib/modules/platform/github/massage-markdown-links.ts#L21-L55
this function should return an array that contains all the links in `prbody`, every object in this array contains from which character the link starts, ends, and what should the link be replaced with.

so after debugging I found that the `start` and `end` numbers that the function returns for each link are incorrect and `what's changed` will not look as expected.
**result:**
https://github.com/MaronHatoum/16921/pull/24

The problem is when running 
https://github.com/renovatebot/renovate/blob/8f8f059d4fdc899a3f4cee805356389d67672561/lib/modules/platform/github/massage-markdown-links.ts#L39

the `tree.value` will be : `pnpm rebuild accepts --store-dir by @​chengcyber in https://github.com/pnpm/pnpm/pull/5036`

![image](https://user-images.githubusercontent.com/98313426/186143072-7e030374-ede0-4a1c-a9f4-bf953ba02b69.png)

but if I copy this value and check it in regex101 I see that there are **some characters does not appear** in `tree.value`(in this case "U+200B")

![image](https://user-images.githubusercontent.com/98313426/186143297-551c24be-a441-4466-b01c-872978403859.png)

so when getting the `startOffset`, there are 6 characters missing and the number will be `startOffset-6`.

**Comment by MaronHatoum** [comment]
blocked by: https://github.com/renovatebot/renovate/issues/17404
